model/modelindexer/bulk_indexer.go

```diff
@@ -135,21 +135,26 @@ func (b *bulkIndexer) writeMeta(item elasticsearch.BulkIndexerItem) {
 	b.jsonw.Reset()
 }
 
-// Flush executes a bulk request if there are any items buffered, and clears out the buffer.
-func (b *bulkIndexer) Flush(ctx context.Context) (elasticsearch.BulkIndexerResponse, error) {
+func (b *bulkIndexer) newBulkRequest() (esapi.BulkRequest, error) {
```

Comment:
We may want to extract `bulkIndexer` out of this package in the future, for reuse in Elastic Agent. To make that possible, we should avoid referring to unexported fields and methods.

What do you think about adding a `bytesFlushed` field to `bulkIndexer`, updating it in the `Flush` method if `req.Do` returns a `nil` error? Then we would expose it with a new `BytesFlushed` method.

Reply:
that sounds like a better approach.